Patches/NewRatkinPlus/ThingDef_Misc/Ratkin_Buildings_Security.xml

```diff
@@ -7,147 +7,222 @@
+			<li Class="PatchOperationReplace">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryBuilding_Cannon"]/thingClass</xpath>
+				<value>
+					<thingClass>CombatExtended.Building_TurretGunCE</thingClass>
+				</value>
+			</li>
+
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryBuilding_Cannon"]/inspectorTabs</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryBuilding_Cannon"]/statBases/AccuracyTouch</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryBuilding_Cannon"]/statBases/AccuracyShort</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryBuilding_Cannon"]/statBases/AccuracyMedium</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryBuilding_Cannon"]/statBases/AccuracyLong</xpath>
+			</li>
+			
+			<li Class="PatchOperationReplace">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryBuilding_Cannon"]/building/turretBurstCooldownTime</xpath>
+				<value>
+					<turretBurstCooldownTime>1</turretBurstCooldownTime>
+				</value>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryWeapon_Cannon"]/building</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryWeapon_Cannon"]/comps/li[@Class="CompProperties_ChangeableProjectile"]</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryWeapon_Cannon"]/statBases/AccuracyTouch</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryWeapon_Cannon"]/statBases/AccuracyShort</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryWeapon_Cannon"]/statBases/AccuracyMedium</xpath>
+			</li>
+			
+			<li Class="PatchOperationRemove">
+				<xpath>Defs/ThingDef[@Name = "RK_BaseArtilleryWeapon_Cannon"]/statBases/AccuracyLong</xpath>
+			</li>
+			
+			<li Class="PatchOperationAdd">
+				<xpath>Defs/ThingDef[defName = "RK_Turret_Cannon_Strait"]</xpath>
+				<value>
+				<statBases/>
+				</value>
+			</li>
+			
+			<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+				<defName>RK_Turret_Cannon_Strait</defName>
+				<statBases>
+					<Mass>20</Mass>
+					<Bulk>20</Bulk>	
+					<SightsEfficiency>1</SightsEfficiency>
+					<ShotSpread>0.12</ShotSpread>
+					<SwayFactor>1.55</SwayFactor>
+					<RangedWeapon_Cooldown>0.36</RangedWeapon_Cooldown>
+				</statBases>
+				<Properties>
+					<recoilAmount>1.13</recoilAmount>
+					<verbClass>CombatExtended.Verb_ShootCE</verbClass>
+					<hasStandardCommand>true</hasStandardCommand>
+					<defaultProjectile>Projectile_RKCannonBall</defaultProjectile>
+					<warmupTime>4.9</warmupTime>
+					<range>50</range>
```

Comment:
Seems somewhat short-ranged, perhaps?

Reply:
I honestly don't really know that sort of point target range a 24 pounder (I think that was what I was using) cannon would have, even if I know that the actual range where it remains fairly lethal againt humans is probably at least a km. Combined with poor ballistics on ball style cannon shot I was inclned to leave it on the lower end.

Reply:
For reference, the Not Only Just A Cannon patch has a range of 58 for most of the medium-ish guns. Figure that's enough to allow for 2 - 3 shots before targets generally get close enough to endanger the crew.

On the note of ballistics, if you used the gun sheet to calculate the gun stats, triple the `ShotSpread` since it's smoothbore.